exchanges/exchange.go

```diff
@@ -319,26 +324,55 @@ func (e *Base) GetSupportedFeatures() FeaturesSupported {
 
 // GetPairFormat returns the pair format based on the exchange and
 // asset type
-func (e *Base) GetPairFormat(assetType asset.Item, requestFormat bool) currency.PairFormat {
+func (e *Base) GetPairFormat(assetType asset.Item, requestFormat bool) (currency.PairFormat, error) {
 	if e.CurrencyPairs.UseGlobalFormat {
 		if requestFormat {
-			return *e.CurrencyPairs.RequestFormat
+			if e.CurrencyPairs.RequestFormat == nil {
+				return currency.PairFormat{},
+					errors.New("global request format is nil")
+			}
+			return *e.CurrencyPairs.RequestFormat, nil
+		}
+
+		if e.CurrencyPairs.ConfigFormat == nil {
+			return currency.PairFormat{},
+				errors.New("global config format is nil")
 		}
-		return *e.CurrencyPairs.ConfigFormat
+		return *e.CurrencyPairs.ConfigFormat, nil
+	}
+
+	ps, err := e.CurrencyPairs.Get(assetType)
+	if err != nil {
+		return currency.PairFormat{}, err
 	}
 
 	if requestFormat {
-		return *e.CurrencyPairs.Get(assetType).RequestFormat
+		if ps.RequestFormat == nil {
+			return currency.PairFormat{},
+				errors.New("asset type request format is nil")
+		}
+		return *ps.RequestFormat, nil
+	}
+
+	if ps.ConfigFormat == nil {
+		return currency.PairFormat{},
+			errors.New("asset type config format is nil")
 	}
-	return *e.CurrencyPairs.Get(assetType).ConfigFormat
+	return *ps.ConfigFormat, nil
 }
 
 // GetEnabledPairs is a method that returns the enabled currency pairs of
 // the exchange by asset type
-func (e *Base) GetEnabledPairs(assetType asset.Item) currency.Pairs {
-	format := e.GetPairFormat(assetType, false)
-	pairs := e.CurrencyPairs.GetPairs(assetType, true)
-	return pairs.Format(format.Delimiter, format.Index, format.Uppercase)
+func (e *Base) GetEnabledPairs(a asset.Item) (currency.Pairs, error) {
```

Comment:
This function is still returning a currency per asset type when they're disabled. eg OKEX websocket enabled, all asset types disabled

Reply:
gross